include/z5/metadata.hxx

```diff
@@ -263,10 +263,20 @@ namespace z5 {
     // helper functions
     //
 
+    inline void writeMetadata(const handle::File & handle, const Metadata & metadata) {
+        auto filePath = handle.path();
+        filePath = metadata.isZarr ? ".zgroup" : "attributes.json";
```

Comment:
This needs to be `filePath /= ...` instead of `filePath = ...`. With `=`, we would set the `filePath` to be `".zgroup"`, but we need to append it (and thus use `\=`).

Reply:
Thanks for your notice. This append helps a lot. Also, I just modified the operator for `dataset` as well as file and group. Let me know if that fits your suggestion, thanks.